localstack/services/apigateway/helpers.py

```diff
@@ -613,36 +613,38 @@ def create_authorizer(path_payload: dict) -> Authorizer:
         security_schemes = path_payload.get("security")
         for security_scheme in security_schemes:
             for security_scheme_name, _ in security_scheme.items():
-                if security_scheme_name in body.get("securityDefinitions"):
+                if security_scheme_name in body.get("securityDefinitions", []):
                     security_config = body.get("securityDefinitions", {}).get(security_scheme_name)
                     aws_apigateway_authorizer = security_config.get(
-                        "x-amazon-apigateway-authorizer"
+                        "x-amazon-apigateway-authorizer", {}
                     )
+                    if not aws_apigateway_authorizer:
```

Comment:
The change is conveyed in this line, if the specification has a no authorizer extensions we can skip it.